### PR TITLE
CDPT-1401: Auto-close expired rejected cases

### PR DESCRIPTION
**Pull request by MeenaModhvadia**
## Description
AS A: Branston user
I WANT: The rejected case to auto close after the external deadline has passed if no information has been received
SO THAT: Rejected cases don't remain open indefinitely

When a case is closed, it is expected to be done by a user. For these automated closures, a system user has been created.

A cronjob will run daily to close expired rejected cases.

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [ ] (4) Branch ready to be merged (not work in progress)
* [ ] (5) No superfluous changes in diff
* [ ] (6) No TODO's without new ticket numbers
* [x] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`
* [ ] (8) Data migration script is created if any of letter templates is changed

### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CDPT-1401

### CODE REVIEW

File: app/jobs/close_rejected_offender_sars_job.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    offender_sars_to_close.each do |offender_sar|
+      offender_sar.update!(current_state: "closed", date_responded: Time.zone.today)
+      offender_sar.save!
+    end
+  end
+
+private
+
+  def offender_sars_to_close
+    _close_offender_sars = Case::SAR::Offender.where(current_state: "rejected")
```
Comment (vertism) [inline app/jobs/close_rejected_offender_sars_job.rb:14]: The assignment here is unnecessary as `_close_offender_sars` is not being used.

File: lib/tasks/close_rejected_offender_sars_task.rake [hunk 2]
```diff
@@ -0,0 +1,9 @@
+namespace :close_rejected_offender_sars do
```
Comment (vertism) [inline lib/tasks/close_rejected_offender_sars_task.rake:1]: Have you tried running this locally?

File: config/kubernetes/development/cronjob-close-rejected-offender-sar.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+apiVersion: batch/v1
```
Comment (vertism) [inline config/kubernetes/development/cronjob-close-rejected-offender-sar.yaml:1]: Currently this cronjob is set to run in the development namespace but nowhere else.

File: lib/tasks/close_rejected_offender_sars_task.rake [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  desc "Close rejected offender sars that were received over the deadline"
+  task update_statuses: :environment do
+    CaseSAROffender.scope: "late".each do |rejected_offender_sar|
```
Comment (vertism) [inline lib/tasks/close_rejected_offender_sars_task.rake:4]: I'm not sure what `CaseSAROffender` is.
You would normally call a scope directly e.g. `Object.scope_name`

File: lib/tasks/close_rejected_offender_sars_task.rake [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+      rejected_offender_sar.update!(current_state: "closed", date_responded: Time.zone.today)
```
Comment (vertism) [inline lib/tasks/close_rejected_offender_sars_task.rake:5]: There is a `CaseClosureService` which should probably be used.

File: lib/tasks/close_rejected_offender_sars_task.rake (hunk 4 above)
Comment (vertism) [reply lib/tasks/close_rejected_offender_sars_task.rake:4]: Shouldn't this only apply to rejected cases?

File: app/jobs/close_rejected_offender_sars_job.rb (hunk 1 above)
Comment (MeenaModhvadia) [reply app/jobs/close_rejected_offender_sars_job.rb]: removed

File: app/jobs/close_rejected_offender_sars_job.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
+class CloseRejectedOffenderSarsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    offender_sars_to_close.each do |offender_sar|
+      offender_sar.update!(current_state: "closed", date_responded: Time.zone.today)
```
[the remark this answers is not in the document]
Comment (MeenaModhvadia) [reply app/jobs/close_rejected_offender_sars_job.rb]: my rake task close_rejected_offender_sars_task.rake uses the CaseClosureService

File: lib/tasks/close_rejected_offender_sars_task.rake (hunk 2 above)
Comment (MeenaModhvadia) [reply lib/tasks/close_rejected_offender_sars_task.rake:1]: Running locally

File: config/kubernetes/development/cronjob-close-rejected-offender-sar.yaml (hunk 3 above)
Comment (MeenaModhvadia) [reply config/kubernetes/development/cronjob-close-rejected-offender-sar.yaml:1]: I have only created this in the kubernettes\development I will copy the file into the other kubernetes environments once working.  Is that ok?

File: lib/tasks/close_rejected_offender_sars_task.rake (hunk 4 above)
Comment (MeenaModhvadia) [reply lib/tasks/close_rejected_offender_sars_task.rake]: Amended the criteria using rejections only and with a scope of late

**Review by lucas-shaw — APPROVED** [review]
LGTM

**Pull request merged**